### PR TITLE
Document avro.schema.literal property use for interpreting table data

### PR DESCRIPTION
**Pull request by Jessie212**
## Description

Document avro.schema.literal property use for interpreting table data

## Additional context and related issues

https://github.com/trinodb/trino/pull/14426

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by mosabua** [comment]
Lets put this on hold until the PR from @m57lyra is merged and then rebase and put it into the correct location.

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -1062,13 +1062,14 @@ format, which has the schema set based on an Avro schema file/literal. Trino is
...
 
-To specify that the Avro schema should be used for interpreting table's data one must use ``avro_schema_url`` table property.
-The schema can be placed remotely in
-HDFS (e.g. ``avro_schema_url = 'hdfs://user/avro/schema/avro_data.avsc'``),
-S3 (e.g. ``avro_schema_url = 's3n:///schema_bucket/schema/avro_data.avsc'``),
-a web server (e.g. ``avro_schema_url = 'http://example.org/schema/avro_data.avsc'``)
-as well as local file system. This URL, where the schema is located, must be accessible from the
-Hive metastore and Trino coordinator/worker nodes.
+To specify that the Avro schema should be used for interpreting table data, use
+the ``avro_schema_url`` or the ``avro.schema.literal`` table property. The
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:1066]: I think this is wrong

 ``avro.schema.literal`` table property 

I think this is a catalog configuration property

and the _ version is either a table property or a session property

**Comment by mosabua** [comment]
Maybe @colebow or @findinpath can clarify the technical aspects..

File: docs/src/main/sphinx/connector/hive.rst (hunk 1 above)
Comment (findinpath) [reply docs/src/main/sphinx/connector/hive.rst:1066]: ``avro.schema.literal`` -> ``avro_schema_literal``

However this paragraph is mostly about `avro_schema_url` and how to eventually retrieve the avro schema of the table.

Comment (findinpath) [reply docs/src/main/sphinx/connector/hive.rst:1066]: @mosabua  `avro.schema.literal` and `avro.schema.url` are Hive specific table properties.

**Comment by m57lyra** [comment]
It is merged @mosabua as of today. Good to go!

**Review by findinpath — APPROVED** [review]

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -868,15 +868,19 @@ format, which has the schema set based on an Avro schema file/literal. Trino is
...
+HDFS (e.g. ``avro_schema_url = 'hdfs://user/avro/schema/avro_data.avsc'``), S3
+(e.g. ``avro_schema_url = 's3n:///schema_bucket/schema/avro_data.avsc'``), a web
+server (e.g. ``avro_schema_url = 'http://example.org/schema/avro_data.avsc'``)
+as well as the local file system. This URL, where the schema is located, must be
+accessible from the Hive metastore and Trino coordinator/worker nodes.
+
+Alternatively, you can use the table property ``avro_schema_literal`` to define
+the Avro schema.
+
+The table created in Trino using ``avro_schema_url`` behaves the same way as a
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive.rst:882]: Shouldn't we have here a reference about `avro_schema_literal` now?

File: docs/src/main/sphinx/connector/hive.rst [hunk 3]
```diff
@@ -868,15 +868,19 @@ format, which has the schema set based on an Avro schema file/literal. Trino is
...
-HDFS (e.g. ``avro_schema_url = 'hdfs://user/avro/schema/avro_data.avsc'``),
-S3 (e.g. ``avro_schema_url = 's3n:///schema_bucket/schema/avro_data.avsc'``),
-a web server (e.g. ``avro_schema_url = 'http://example.org/schema/avro_data.avsc'``)
-as well as local file system. This URL, where the schema is located, must be accessible from the
-Hive metastore and Trino coordinator/worker nodes.
-
-The table created in Trino using ``avro_schema_url`` behaves the same way as a Hive table with ``avro.schema.url`` or ``avro.schema.literal`` set.
+To specify that the Avro schema should be used for interpreting table data, use
+the ``avro_schema_url`` table property. The schema can be placed remotely in
+HDFS (e.g. ``avro_schema_url = 'hdfs://user/avro/schema/avro_data.avsc'``), S3
```
Comment (findinpath) [inline docs/src/main/sphinx/connector/hive.rst:873]: optional: A listing could be a bit easier to scan by the reader.

**Review by mosabua — APPROVED** [review]
Looks good.

**Comment by mosabua** [comment]
Could @ebyhr @hashhar or @findepi review and merge please

**Review by ebyhr — APPROVED** [review]

**Pull request merged**